src/amber/support/params.cr

```diff
@@ -0,0 +1,42 @@
+require "./predicates"
+
+class Check
+  def self.it(match)
+    match.nil? ? false : true
+  end
+end
+
+# Monkey patch the
+class String
+  include Amber::Support::Predicates::String
+end
+
+abstract struct Number
+  include Amber::Support::Predicates::Number
+end
+
+module Amber
+  module Support
+    class Params
+      property params
+      property valid : Bool = false
```

Comment:
you can just `property valid = false`, the compiler will guess from `false` that `valid` is a boolean